src/controllers/generate.controller.ts

```diff
@@ -37,13 +37,22 @@ export class GenerateController implements Controller {
       tmpDir = await this.archiverService.createTempDirectory();
       const { asyncapi, template, parameters } = req.body;
 
-      await this.generatorService.generate(
-        req.parsedDocument,
-        template,
-        parameters,
-        tmpDir,
-        prepareParserConfig(req),
-      );
+      try {
+        await this.generatorService.generate(
+          asyncapi,
+          template,
+          parameters,
+          tmpDir,
+          prepareParserConfig(req),
+        );
+      } catch (genErr: unknown) {
+        return next(new ProblemException({
+          type: 'internal-generator-error',
+          title: 'Internal Generator error',
+          status: 422,
```

Comment:
Wondering why we would consider any error a 422. I understand that if a parameter is not expected, to return such 422 but not when, for example, the generator fails due to a bug or similar.

Reply:
Because we don't have (like in parser) exposed errors types. I agree that not every error should be consider as `422` but I think that it's better than `500`, where it shouldn't be a server error, but internal error of tool (not know for us).

Do you have any idea for other (better) error code?

Reply:
422 is giving the responsibility to the user to change the request. Is telling users the request is semantically invalid. If we can't ensure all errors are because of semantically invalid provided values, then I think we should definitely throw a 500 error.

Reply:
Ok, I will change it to the 500 👌🏼

Reply:
Changed :) Thanks!